packages/default/scss/switch/_theme.scss

```diff
@@ -47,73 +15,124 @@ $switch-active-shadow: 0 0 0 2px rgba(0, 0, 0, .06) !default;
+        }
 
-            &:active {
-                @include appearance( active-primary-button );
+        &:hover,
+        &.k-state-hover {
+            .k-switch-container {
+                @include fill(
+                    $switch-on-hovered-text,
+                    $switch-on-hovered-bg,
+                    $switch-on-hovered-border,
+                    $switch-on-hovered-gradient
+                );
             }
-        }
 
-        &:hover {
             .k-switch-handle {
-                @include appearance( hovered-primary-button );
+                @include fill(
+                    $switch-on-handle-hovered-text,
+                    $switch-on-handle-hovered-bg,
+                    $switch-on-handle-hovered-border,
+                    $switch-on-handle-hovered-gradient
+                );
             }
         }
+
+        &:focus,
+        &.k-state-focused {
+            @include box-shadow( $switch-on-focused-shadow );
+        }
+
+        .k-switch-label-off {
+            color: transparent;
+        }
     }
 
     .k-switch-off {
+        .k-switch-container {
+            @include fill(
+                $switch-off-text,
+                $switch-off-bg,
+                $switch-off-border,
+                $switch-off-gradient
+            );
+        }
+
         .k-switch-handle {
-            background-clip: padding-box;
+            @include fill(
+                $switch-off-handle-text,
+                $switch-off-handle-bg,
+                $switch-off-handle-border,
+                $switch-off-handle-gradient
+            );
+            @include box-shadow( $switch-off-handle-shadow );
         }
-        
-        &:hover {
+
+        &:hover,
+        &.k-state-hover {
+            .k-switch-container {
+                @include fill(
+                    $switch-off-hovered-text,
+                    $switch-off-hovered-bg,
+                    $switch-off-hovered-border,
+                    $switch-off-hovered-gradient
+                );
+            }
+
             .k-switch-handle {
-                @include appearance( hovered-button );
+                @include fill(
+                    $switch-off-handle-hovered-text,
+                    $switch-off-handle-hovered-bg,
+                    $switch-off-handle-hovered-border,
+                    $switch-off-handle-hovered-gradient
+                );
             }
         }
-    }
-
-    .k-switch-label-on {
-        color: $switch-on-color;
-        background-color: transparent;
-    }
 
-    .k-switch-label-off {
-        color: $switch-off-color;
-    }
-
-    .k-switch {
         &:focus,
         &.k-state-focused {
-            outline: none;
-            box-shadow: $switch-focus-shadow;
+            @include box-shadow( $switch-off-focused-shadow );
         }
 
-        &:active {
-            box-shadow: $switch-active-shadow;
+        .k-switch-label-on {
+            color: transparent;
```

Comment:
do we need to assign a variable here for customization?

Reply:
No, we don't, this is the color of the label in the opposite state. For example, when the state is "OFF" we always want that the "ON" label is not present(transparent);